infrastructure/src/main/java/org/corfudb/infrastructure/log/Compression/ZSTDCompression.java

```diff
@@ -0,0 +1,67 @@
+package org.corfudb.infrastructure.log.Compression;
+
+import com.github.luben.zstd.Zstd;
+
+import java.nio.ByteBuffer;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class ZSTDCompression implements Codec {
+
+    private static final int DEFAULT_COMPRESSION_LEVEL = 3;
+
+    private static ZSTDCompression INSTANCE = new ZSTDCompression();
+
+    public static ZSTDCompression getInstance() {
+        return INSTANCE;
+    }
+
+    @Override
+    public ByteBuffer compress(ByteBuffer uncompressed) {
+        checkNotNull(uncompressed);
```

Comment:
why not `Objects.requireNonNull()` ?

Reply:
Bet.

Reply:
using a std library vs 3rd party library for the same behavior. You should always choose the std lib

Reply:
I believe I was misunderstood. Bet means for sure. Thanks for explanation.

Reply:
oh man )) ok

Reply:
done